sqlstats/sqlstats.go

```diff
@@ -164,7 +165,15 @@ func (t *Tracer) Query(prepCtx context.Context, id sqliteh.TraceConnID, query st
 	stats := t.queryStats(query)
 
 	atomic.AddInt64(&stats.Count, 1)
-	atomic.AddInt64(&stats.Duration, int64(duration))
+
+	newDuration := stats.Duration.Load()
+	if newDuration == nil {
+		newDuration = &duration
+	} else {
+		*newDuration = *newDuration + duration
+	}
+	stats.Duration.Store(newDuration)
```

Comment:
Separating the load and store means this is no-longer an atomic Add, like AddInt64 is. If you want to make `stats.Duration` a true time.Duration then you can just convert it in the eyes of the AddInt64 call: https://go.dev/play/p/1Xeiyf3l1Zt

```go
// If stats.Duration is a time.Duration you can do:
atomic.AddInt64((*int64)(&stats.Duration), int64(duration))
```

And then you won't need to call Load() everywhere else.